Share serialization of FontCustomPlatformData

donny-dont opened the pull request:
#### ffd9887f8ea5b790c4e2ec7d774832ad1330ecbc
<pre>
Share serialization of FontCustomPlatformData
<a href="https://bugs.webkit.org/show_bug.cgi?id=255615">https://bugs.webkit.org/show_bug.cgi?id=255615</a>

Reviewed by Alex Christensen.

All platforms have an implementation of `createFontCustomPlatformData`
so just use that to serialize `FontCustomPlatformData`.

* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.mm:
(IPC::ArgumentCoder&lt;WebCore::FontCustomPlatformData&gt;::encodePlatformData): Deleted.
(IPC::ArgumentCoder&lt;WebCore::FontCustomPlatformData&gt;::decodePlatformData): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;WebCore::FontCustomPlatformData&gt;::encode):
(IPC::ArgumentCoder&lt;FontCustomPlatformData&gt;::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.h:
* Source/WebKit/Shared/freetype/WebCoreArgumentCodersFreeType.cpp: Renamed from Source/WebKit/Shared/playstation/WebCoreArgumentCodersPlayStation.cpp.
(IPC::ArgumentCoder&lt;WebCore::Font&gt;::encodePlatformData):
(IPC::ArgumentCoder&lt;WebCore::Font&gt;::decodePlatformData):
(IPC::ArgumentCoder&lt;WebCore::FontPlatformData::Attributes&gt;::encodePlatformData):
(IPC::ArgumentCoder&lt;WebCore::FontPlatformData::Attributes&gt;::decodePlatformData):
* Source/WebKit/Shared/soup/WebCoreArgumentCodersSoup.cpp:
(IPC::ArgumentCoder&lt;Font&gt;::encodePlatformData): Deleted.
(IPC::ArgumentCoder&lt;Font&gt;::decodePlatformData): Deleted.
(IPC::ArgumentCoder&lt;WebCore::FontCustomPlatformData&gt;::encodePlatformData): Deleted.
(IPC::ArgumentCoder&lt;WebCore::FontCustomPlatformData&gt;::decodePlatformData): Deleted.
(IPC::ArgumentCoder&lt;WebCore::FontPlatformData::Attributes&gt;::encodePlatformData): Deleted.
(IPC::ArgumentCoder&lt;WebCore::FontPlatformData::Attributes&gt;::decodePlatformData): Deleted.
* Source/WebKit/Shared/win/WebCoreArgumentCodersWin.cpp:
(IPC::ArgumentCoder&lt;WebCore::FontCustomPlatformData&gt;::encodePlatformData): Deleted.
(IPC::ArgumentCoder&lt;WebCore::FontCustomPlatformData&gt;::decodePlatformData): Deleted.
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:

Canonical link: <a href="https://commits.webkit.org/263103@main">https://commits.webkit.org/263103@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/101b188a6ab1e51bb44e171c3911783db6c73c9a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3601 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3642 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3790 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/5027 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3865 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/3742 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3690 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/3123 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3645 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/3887 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/3245 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4879 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/1393 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/3218 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/3302 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/3194 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/3277 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/4645 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3654 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/2956 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/3189 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/3218 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/881 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/3220 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3469 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 reviewed (approved):
A step in the right direction.  Would be nice to use *.serialization.in files, but that can be a second step.

donny-dont committed:
Share serialization of FontCustomPlatformData
https://bugs.webkit.org/show_bug.cgi?id=255615

Reviewed by Alex Christensen.

All platforms have an implementation of `createFontCustomPlatformData`
so just use that to serialize `FontCustomPlatformData`.

* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.mm:
(IPC::ArgumentCoder<WebCore::FontCustomPlatformData>::encodePlatformData): Deleted.
(IPC::ArgumentCoder<WebCore::FontCustomPlatformData>::decodePlatformData): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder<WebCore::FontCustomPlatformData>::encode):
(IPC::ArgumentCoder<FontCustomPlatformData>::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.h:
* Source/WebKit/Shared/freetype/WebCoreArgumentCodersFreeType.cpp: Renamed from Source/WebKit/Shared/playstation/WebCoreArgumentCodersPlayStation.cpp.
(IPC::ArgumentCoder<WebCore::Font>::encodePlatformData):
(IPC::ArgumentCoder<WebCore::Font>::decodePlatformData):
(IPC::ArgumentCoder<WebCore::FontPlatformData::Attributes>::encodePlatformData):
(IPC::ArgumentCoder<WebCore::FontPlatformData::Attributes>::decodePlatformData):
* Source/WebKit/Shared/soup/WebCoreArgumentCodersSoup.cpp:
(IPC::ArgumentCoder<Font>::encodePlatformData): Deleted.
(IPC::ArgumentCoder<Font>::decodePlatformData): Deleted.
(IPC::ArgumentCoder<WebCore::FontCustomPlatformData>::encodePlatformData): Deleted.
(IPC::ArgumentCoder<WebCore::FontCustomPlatformData>::decodePlatformData): Deleted.
(IPC::ArgumentCoder<WebCore::FontPlatformData::Attributes>::encodePlatformData): Deleted.
(IPC::ArgumentCoder<WebCore::FontPlatformData::Attributes>::decodePlatformData): Deleted.
* Source/WebKit/Shared/win/WebCoreArgumentCodersWin.cpp:
(IPC::ArgumentCoder<WebCore::FontCustomPlatformData>::encodePlatformData): Deleted.
(IPC::ArgumentCoder<WebCore::FontCustomPlatformData>::decodePlatformData): Deleted.
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:

Canonical link: https://commits.webkit.org/263103@main
